core/control-plane/contract-core/src/main/java/org/eclipse/edc/connector/contract/validation/ContractValidationServiceImpl.java

```diff
@@ -78,77 +83,43 @@
         }
 
         var agent = agentService.createFor(token);
 
-        var consumerIdentity = agent.getIdentity();
-        if (consumerIdentity == null) {
-            return failure("Invalid consumer identity");
-        }
-
-        var contractDefinition = contractDefinitionResolver.definitionFor(agent, contractId.definitionPart());
-        if (contractDefinition == null) {
-            return failure(
-                    "The ContractDefinition with id %s either does not exist or the access to it is not granted.");
-        }
-
-        var targetAsset = assetIndex.findById(offer.getAssetId());
-        if (targetAsset == null) {
-            return failure("Invalid target: " + offer.getAssetId());
-        }
-
-        // verify that the asset in the offer is actually in the contract definition
-        var testCriteria = new ArrayList<>(contractDefinition.getSelectorExpression().getCriteria());
-        testCriteria.add(new Criterion(Asset.PROPERTY_ID, "=", offer.getAssetId()));
-        if (assetIndex.countAssets(testCriteria) <= 0) {
-            return failure("Asset ID from the ContractOffer is not included in the ContractDefinition");
-        }
-
-        // if policy target is null, default to the asset, otherwise validate it
-        var policyTarget = offer.getPolicy().getTarget() != null ? offer.getPolicy().getTarget() : targetAsset.getId();
-        if (!targetAsset.getId().equals(policyTarget)) {
-            return failure(format("Contract offer asset '%s' does not match policy target: %s", offer.getAssetId(), policyTarget));
-        }
-
-        var contractPolicyDef = policyStore.findById(contractDefinition.getContractPolicyId());
-        if (contractPolicyDef == null) {
-            return failure(format("Policy %s not found", contractDefinition.getContractPolicyId()));
-        }
+        var result = validateInitialOffer(contractId, agent);
+
+        return result.map(r ->
+                new ValidatedConsumerOffer(agent.getIdentity(),
+                        ContractOffer.Builder.newInstance()
+                                .id(offer.getId())
+                                .assetId(contractId.assetIdPart())
+                                .providerId(participantId)
+                                .policy(r.getPolicy())
+                                .contractStart(offer.getContractStart())
+                                .contractEnd(offer.getContractEnd())
+                                .build())
+        );
+    }
 
-        var offerValidity = ChronoUnit.SECONDS.between(offer.getContractStart(), offer.getContractEnd());
-        if (offerValidity != contractDefinition.getValidity()) {
-            return failure(format("Offer validity %ss does not match contract definition validity %ss", offerValidity, contractDefinition.getValidity()));
+    @Override
+    public @NotNull Result<ValidatedConsumerOffer> validateInitialOffer(ClaimToken token, String offerId) {
```

Comment:
I couldn't find any tests for this method. Even though the behaviour is almost identical, IMO there should still be tests - if only to guard against future changes in behaviour.

Reply:
let me put one in